include/revng/ADT/Concepts.h

```diff
@@ -4,44 +4,178 @@
 // This file is distributed under the MIT License. See LICENSE.md for details.
 //
 
+#include <array>
 #include <concepts>
 #include <iterator>
+#include <string>
+#include <string_view>
 #include <type_traits>
 
-template<typename T, typename U>
-concept convertible_to = std::is_convertible_v<T, U>;
-
-template<typename T>
-concept NotVoid = not std::is_void_v<T>;
+//
+// The concepts from the STL.
+// TODO: remove these after updating the libc++ version.
+//
 
+// clang-format off
 template<class Derived, class Base>
-concept DerivesFrom = std::is_base_of_v<Base, Derived>;
+concept derived_from = std::is_base_of_v<Base, Derived>
+                       && std::is_convertible_v<const volatile Derived *,
+                                                const volatile Base *>;
+// clang-format on
```

Comment:
I'm starting to think that this whole file should be wrapped in `// clang-format off`

Reply:
Yeah, I thought so as well, but there are also a couple of non-concept things in here - and there can never be too many consistency checks.